src/app/application/components/app-collection/AppCollection.vue

```diff
@@ -54,41 +59,117 @@ type ChangeValue = {
 }
 
 const props = withDefaults(defineProps<{
-  total: number,
+  selectedRowId?: string | null
+  total?: number,
   pageNumber: number,
   pageSize: number,
   items: unknown[] | undefined,
   error: Error | undefined,
 }>(), {
+  selectedRowId: null,
   total: 0,
 })
 
 const emit = defineEmits<{
   (e: 'change', value: ChangeValue): void
+  (event: 'row:click', row: any): void
+  (event: 'row:selected', selectedRowId: string): void
 }>()
 
 const slots = useSlots()
 
+const currentSelectedRowId = ref<string | null>(null)
 const items = ref<unknown[] | undefined>(props.items)
 const cacheKey = ref<number>(0)
+
+const remainingSlots = computed(() => {
+  const { toolbar, ...remainingSlots } = slots
+
+  return remainingSlots
+})
+
 watch(() => props.items, () => {
   cacheKey.value++
   items.value = props.items
 })
-const click = (e: MouseEvent) => {
-  const $tr = (e.target as HTMLElement).closest('tr')
-  if ($tr) {
-    const $a = $tr.querySelector('a')
-    if ($a !== null) {
-      $a.click()
+
+watch(() => props.items, function (items) {
+  if (currentSelectedRowId.value !== null) {
+    return
+  }
+
+  if (Array.isArray(items) && items.length > 0) {
+    if (props.selectedRowId !== null) {
+      // Use `props.selectedRowId` if it actually exists in `props.items`
+      const hasItem = items.find((item) => item && typeof item === 'object' && 'id' in item && item.id === props.selectedRowId)
+
+      if (hasItem) {
+        currentSelectedRowId.value = props.selectedRowId
+      }
+    } else {
+      // Use the first item as a fallback.
+      const item = items[0]
+
+      if (item && typeof item === 'object' && 'id' in item && typeof item.id === 'string') {
```

Comment:
**Comment**: Not in love with the necessity for that degree of type-checking this value.

I’d rather type `props.items` as a list of objects that at the very least have an `id` property (so we can standardize on _one_ behavior for selecting an active row). But I didn’t want to add this here because it would increase the scope.